### ISSUE TITLE
[SR-13560] swiftpm: SWIFTPM_MODULE_BUNDLE returning nil

### ISSUE THREAD

**Issue by thomasvl** [apple/swift-package-manager#4500]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-13560      |
   |Radar             | rdar://problem/69090207         |
   |Original Reporter | @thomasvl      |
   |Type              | Bug    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 5         |
   |Component/s       | Package Manager    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 54f1a590f07af2575ba60b5ea19e1321

  </details>

**is duplicated by**:  
* [SR-13714](https://bugs.swift.org/browse/SR-13714) 'Swift test' fails when including package resources  

**Issue Description:**

An all ObjC package is getting `nil` from `SWIFTPM_MODULE_BUNDLE` when using `swift test`, but if the `Package.swift` is directly opened in Xcode 12, then the tests pass as the bundle is returned.

Project: <https://github.com/google/gtm-session-fetcher>  
Commit attempt to move the 5.3 Resource support: <https://github.com/thomasvl/gtm-session-fetcher/commit/f38d107e93ebc4571b9985665ad5e324ed7fd924>

Raised in <https://forums.swift.org/t/5-3-resources-support-not-working-on-with-swift-test/40381>

**Comment by mokagio** [comment]
We run into this issue, too, with 5.5. It can be verified by checking out [this commit](https://github.com/Automattic/Automattic-Tracks-iOS/pull/206/commits/bc86865be9e89fd32c0781469d629fd5d0d01a68).

For what is worth, `swift tests` _worked_ with the 5.3 tools.

**Comment by neonichu** [comment]
Looks like the implementation is

```
NSBundle* \(target.c99name)_SWIFTPM_MODULE_BUNDLE() {
            NSURL *bundleURL = [[[NSBundle mainBundle] bundleURL] URLByAppendingPathComponent:@"\(bundleBasename)"];
            return [NSBundle bundleWithURL:bundleURL];
        }
```

IIRC, the main bundle when executing tests is the test runner, so it makes sense that this would return nil.

**Comment by ncooke3** [comment]
@neonichu, I also ran into this while manually testing #5919. I have proposed a fix for this issue in #5921.
